File: backend/app/schemas/reporting.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class ReportingSummary(BaseModel):
    trade_count: int
    realized_pnl: float
    win_rate: float
    cumulative_return: float
    profit_factor: float
    max_drawdown: float
    avg_win: float
    avg_loss: float
    annualized_return_pct: float | None = None
    annualized_volatility_pct: float | None = None
    sharpe_ratio: float | None = None
    calmar_ratio: float | None = None


class EquityCurvePoint(BaseModel):
    label: str
    total_equity: float


class PeriodStat(BaseModel):
    period: str
    trade_count: int
    realized_pnl: float
    ending_equity: float


class EventLogRead(BaseModel):
    id: int
    tenant_id: str
    user_id: int | None = None
    account_id: int | None = None
    event_type: str
    symbol: str | None = None
    occurred_at: datetime
    strategy_id: int | None = None
    strategy_run_id: int | None = None
    order_id: int | None = None
    order_event_id: int | None = None
    trade_id: int | None = None
    position_id: int | None = None
    equity_snapshot_id: int | None = None
    correlation_id: str | None = None
    risk_rule_version: str | None = None
    payload: dict[str, Any]

File: frontend/src/types/reporting.ts
export interface ReportingSummary {
  trade_count: number
  realized_pnl: number
  win_rate: number
  cumulative_return: number
  profit_factor: number
  max_drawdown: number
  avg_win: number
  avg_loss: number
  annualized_return_pct: number | null
  annualized_volatility_pct: number | null
  sharpe_ratio: number | null
  calmar_ratio: number | null
}

export interface TotalAssetCurvePoint {
  label: string
  total_equity: number
}

export type EquityCurvePoint = TotalAssetCurvePoint

export interface PeriodStat {
  period: string
  trade_count: number
  realized_pnl: number
  ending_equity: number
}

export interface ReportingEvent {
  id: number
  tenant_id: string
  user_id: number | null
  account_id: number | null
  event_type: string
  symbol: string | null
  occurred_at: string
  strategy_id: number | null
  strategy_run_id: number | null
  order_id: number | null
  order_event_id: number | null
  trade_id: number | null
  position_id: number | null
  equity_snapshot_id: number | null
  correlation_id: string | null
  risk_rule_version: string | null
  payload: Record<string, unknown>
}
